src/Config.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+/*
+ * This file is part of the Flintstone package.
+ *
+ * (c) Jason M <emailfire@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Flintstone;
+
+use Doctrine\Common\Cache\Cache;
+use Doctrine\Common\Cache\ArrayCache;
+use Doctrine\Common\Cache\ClearableCache;
+use Flintstone\Formatter\FormatterInterface;
+use Flintstone\Formatter\SerializeFormatter;
+
+class Config
+{
+    /**
+     * Default config.
+     *
+     * @var array
+     */
+    protected $config = array(
+        'dir' => '',
+        'ext' => '.dat',
+        'gzip' => false,
+        'cache' => true,
+        'formatter' => null,
+        'swap_memory_limit' => 2097152,    // 2MB
+    );
+
+    /**
+     * Constructor.
+     *
+     * @param array $config
+     */
+    public function __construct(array $config = array())
+    {
```

Comment:
The reason it's done this way is so that each config that is provided run's through it's own setter method which does the validation. The protected $config contains the default value's if nothing is passed through the constructor.